lib/src/spec/common.dart

```diff
@@ -74,31 +75,189 @@ abstract class StatsMessageTraffic {
   StatsMessageTypes webhook;
 }
 
+/// A class providing parameters of a token request.
+///
+/// Parameters for a token request
+///
+/// [Auth.authorize], [Auth.requestToken] and [Auth.createTokenRequest]
+/// accept an instance of TokenParams as a parameter
+///
+/// spec: https://docs.ably.io/client-lib-development-guide/features/#TK1
 class TokenParams {
+
+  /// Capability of the token.
+  ///
+  /// If the token request is successful, the capability of the
+  /// returned token will be the intersection of this capability
+  /// with the capability of the issuing key.
+  ///
+  /// Ref: TK2b
   String  capability;
+
+  /// A clientId to associate with this token.
+  ///
+  /// The generated token may be used to authenticate as this clientId.
+  ///
+  /// Ref: TK2c
   String clientId;
+
+  /// An opaque nonce string of at least 16 characters to ensure uniqueness.
+  ///
+  /// Timestamps, in conjunction with the nonce,
+  /// are used to prevent requests from being replayed
+  ///
+  /// ref: TK2d
   String nonce;
+
+  /// The timestamp (in millis since the epoch) of this request.
+  ///
+  ///	Timestamps, in conjunction with the nonce, are used to prevent
+  ///	token requests from being replayed.
+  ///
+  /// ref: TK2d
   DateTime timestamp;
+
+  /// Requested time to live for the token.
+  ///
+  /// If the token request is successful, the TTL of the returned
+  /// token will be less than or equal to this value depending on
+  /// application settings and the attributes of the issuing key.
+  ///
+  /// 0 means Ably will set it to the default value
+  ///
+  /// Ref: TK2a
   int ttl;
+
+  TokenParams({
+    this.capability,
+    this.clientId,
+    this.nonce,
+    this.timestamp,
+    this.ttl
+  });
+
 }
 
+/// TokenDetails is a type containing the token request
```

Comment:
This comment should be improved. 
Using the name of the commented item in the comment usually leads to comments that do not add much value. Also that `TokenDetail` is a type is not useful, because `class TokenDetails` tells that already.

My suggestion

```dart
/// Response to a REST requestToken request.
```

Reply:
I think we better say:

"An encapsulating class TokenDetails acquired by making a request to `requestToken` endpoint"

Reply:
```
Response to a `requestToken` endpoint
```
seems to be enough for me, except that's I'm not sure what `endpoint` means here. Perhaps 
```
Response to a `requestToken` request
```
would be better. `ecapsulating class` and `TokenDetails` doesn't add any value in my opinion.

Reply:
a162720d5a628526e5257889ace180d9fbfe0a48